src/operations.jl

```diff
@@ -1,3 +1,77 @@
+==(c1::Lab, c2::Lab) = c1.l == c2.l && c1.a == c2.a && c1.b == c2.b
+==(c1::Luv, c2::Luv) = c1.l == c2.l && c1.u == c2.u && c1.v == c2.v
+==(c1::LCHab, c2::LCHab) = c1.l == c2.l && c1.c == c2.c && c1.h == c2.h
+==(c1::LCHuv, c2::LCHuv) = c1.l == c2.l && c1.c == c2.c && c1.h == c2.h
+==(c1::DIN99, c2::DIN99) = c1.l == c2.l && c1.a == c2.a && c1.b == c2.b
+==(c1::DIN99d, c2::DIN99d) = c1.l == c2.l && c1.a == c2.a && c1.b == c2.b
+==(c1::DIN99o, c2::DIN99o) = c1.l == c2.l && c1.a == c2.a && c1.b == c2.b
+==(c1::LMS, c2::LMS) = c1.l == c2.l && c1.m == c2.m && c1.s == c2.s
+==(c1::YIQ, c2::YIQ) = c1.y == c2.y && c1.i == c2.i && c1.q == c2.q
+==(c1::YCbCr, c2::YCbCr) = c1.y == c2.y && c1.cb == c2.cb && c1.cr == c2.cr
+
+==(x::AbstractGray, y::AbstractGray) = gray(x) == gray(y)
+==(x::Number, y::AbstractGray) = x == gray(y)
+==(x::AbstractGray, y::Number) = ==(y, x)
+
+function ==(x::TransparentColor, y::TransparentColor)
+    color(x) == color(y) && alpha(x) == alpha(y)
+end
+
+
+struct BoolTuple end
+@inline BoolTuple(args::Bool...) = (args...,)
+
+function _isapprox(a::Colorant, b::Colorant; kwargs...)
+    componentapprox(x, y) = isapprox(x, y; kwargs...)
+    all(ColorTypes._mapc(BoolTuple, componentapprox, a, b))
+end
+isapprox(a::C, b::C; kwargs...) where {C<:Colorant} =
+    _isapprox(a, b; kwargs...)
+isapprox(a::Colorant, b::Colorant; kwargs...) =
+    _isapprox(base_colorant_type(a), base_colorant_type(b), a, b; kwargs...)
+_isapprox(::Type{C}, ::Type{C}, a, b; kwargs...) where {C<:Colorant} =
+    _isapprox(a, b; kwargs...)
+_isapprox(::Type{<:AbstractRGB}, ::Type{<:AbstractRGB}, a, b; kwargs...) =
+    _isapprox(RGB(a), RGB(b); kwargs...)
+_isapprox(::Type{<:AbstractGray}, ::Type{<:AbstractGray}, a, b; kwargs...) =
+    isapprox(gray(a), gray(b); kwargs...)
+_isapprox(TA::Type, TB::Type, a, b; kwargs...) = false
+isapprox(x::Number, y::AbstractGray; kwargs...) =
+    isapprox(x, gray(y); kwargs...)
+isapprox(x::AbstractGray, y::Number; kwargs...) =
+    isapprox(y, x; kwargs...)
+
+zero(::Type{C}) where {C<:ColorN{1}} = C(0)
+zero(::Type{Gray24})                 = Gray24(zero(N0f8))
+zero(::Type{C}) where {C<:ColorN{3}} = C(0, 0, 0)
+# It's not clear whether zero(TransparentColor) should use alpha = 0 or alpha = 1
+zero(c::Colorant)                    = zero(typeof(c))
+
+oneunit(::Type{C}) where {C<:ColorN{1}}       = C(1)
+oneunit(::Type{Gray24})                       = Gray24(oneunit(N0f8))
+# It's not entirely clear what `oneunit` means for most Color3s, but for AbstractRGB it's OK
+oneunit(::Type{C}) where {C<:AbstractRGB}     = C(1, 1, 1)
+oneunit(::Type{C}) where {C<:TransparentGray} = C(oneunit(color_type(C)), 1)
+oneunit(::Type{C}) where {C<:TransparentRGB}  = C(oneunit(color_type(C)), 1)
+oneunit(c::Colorant)                          = oneunit(typeof(c))
+
+function Base.one(::Type{C}) where {C<:Gray}
+    Base.depwarn("one($C) will soon switch to returning 1; you might need to switch to `oneunit`", :one)
+    C(1)
+end
```

Comment:
One more method?
```julia
one(c::Gray) = one(typeof(c))
```

Reply:
Oh shoot I forgot about #175. Sorry!

WRT 1, since this package doesn't support multiplication, what should be returned for the multiplicative identity?

Reply:
Hmmm, are you indicating that `one(::Type{<:Gray}) = C(1)` should be moved to ColorVectorSpace without depwarn?

Reply:
actually, why don't we add `white`/`whites`/`black`/`blacks` as the direct replacements of `one`/`ones`/`zero`/zeros`?

Reply:
For example, what should `white` or `black` return in `HSL`?
After all, if they are obvious, abstraction is not really helpful.

Reply:
`HSL(gamutmin(HSL)...)` and `HSL(gamutmax(HSL)...)` looks good to me.

The point of introducing these two convenience function is, if people ask you what's the difference between `one`/`oneunit` and why `zeros` works while `ones` doesn't, you could save yourself some efforts and tell them to use `blacks` and `whites`. Not everyone is color expert and people from other languages do miss `ones` when they actually want `whites`.

Reply:
I consider `zeros` and `ones` to be array initializers, so I don't think anything requiring additional documentation is appropriate for the "identity". We implicitly or informally expect identities to be unique.
Therefore, I think there may be `black`/`white` which returns "an example" of black / white, but `blacks`/`whites` is never better than `zeros`/`ones`.

BTW, unfortunately, we already have the `red`/`green`/`blue`. The `black`/`white` will introduce inconsistencies.:ghost: